<a href="https://colab.research.google.com/github/priscygpc/priscygpc/blob/main/An%C3%A1lise_de_Dados_em_Python_Conte%C3%BAdo_do_Curso_basico_Python_ipynb_Priscila_Paiva_Cavalcante.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Passo a passo de Solução do Desafio 


In [47]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


###Lógica de programação
###Passo 0 - Entender o desafio que você quer resolver
###Passo 1 - Percorrer todos os arquivos da pasta base de dados (Pasta Vendas)
o que é uma biblioteca? #**os** é uma bilbioteca - criada para trabalhar percorre pastas de arquivos.

import os
import pandas as pd

lista_arquivo = os.listdir()

###Passo 2 - Importar as bases de dados de vendas
for arquivo in lista_arquivo
for variavel in lista_items:
###Importar Arquivo
import pandas (para importar arquivos)
de forma dinânica = 
###Passo 3 - Tratar/Compilar as bases de dados
###Passo 4 - Calcular o produto mais vendido (em quantidade)
###Passo 5 - Calcular o produto que mais faturou ( em faturamento)
###Passo 6 - Calcular a loja/cidade que mais (em faturamento) - criar um gráfico/dashboard

In [48]:
###Lógica de programação
###Passo 0 - Entender o desafio que você quer resolver
###Passo 1 - Percorrer todos os arquivos da pasta base de dados (Pasta Vendas)
#o que é uma biblioteca? #**os** é uma bilbioteca - criada para trabalhar percorre pastas de arquivos.

import os
import pandas as pd

lista_arquivo = os.listdir('/content/drive/MyDrive/Vendas/')
display(lista_arquivo)

['Devolucoes - São Paulo.csv',
 'Vendas - Porto Alegre.csv',
 'Devolucoes - Recife.csv',
 'Devolucoes - Porto Alegre.csv',
 'Vendas - Fortaleza.csv',
 'Devolucoes - Goiás.csv',
 'Vendas - Salvador.csv',
 'Devolucoes - Rio de Janeiro.csv',
 'Devolucoes - Salvador.csv',
 'Devolucoes - Curitiba.csv',
 'Vendas - Rio de Janeiro.csv',
 'Devolucoes - Fortaleza.csv',
 'Vendas - São Paulo.csv',
 'Vendas - Belo Horizonte.csv',
 'Devolucoes - Belo Horizonte.csv',
 'Vendas - Recife.csv',
 'Vendas - Goiás.csv',
 'Vendas - Curitiba.csv']

In [49]:
tabela_total = pd.DataFrame()

#Passo 2 - Importar as bases de dados de vendas
#for arquivo in lista_arquivo for variavel in lista_items:
#Importar Arquivo
#lista_arquivo = os.listdir('/content/drive/MyDrive/Vendas/')
#display(lista_arquivo)

for arquivo in lista_arquivo:

 #Se vendas estiver no nome do arquivo , então:
  if "Vendas" in arquivo:

  #print(f'/content/drive/MyDrive/Vendas/{arquivo}')
  #Importar arquivo

    tabela = pd.read_csv(f"/content/drive/MyDrive/Vendas/{arquivo}") 
    tabela_total = tabela_total.append(tabela)
    print(arquivo)
display(tabela_total)

Vendas - Porto Alegre.csv
Vendas - Fortaleza.csv
Vendas - Salvador.csv
Vendas - Rio de Janeiro.csv
Vendas - São Paulo.csv
Vendas - Belo Horizonte.csv
Vendas - Recife.csv
Vendas - Goiás.csv
Vendas - Curitiba.csv


Unnamed: 0     SKU     Produto  ...          Loja Preco Unitario Unnamed: 8
0            14  HL4379   Televisão  ...  Porto Alegre           2500        NaN
1            34  HL1148      Câmera  ...  Porto Alegre           2100        NaN
2            43  HL1148      Câmera  ...  Porto Alegre           2100        NaN
3            49  HL8851    Notebook  ...  Porto Alegre           3500        NaN
4            52  HL4379   Televisão  ...  Porto Alegre           2500        NaN
..          ...     ...         ...  ...           ...            ...        ...
714        9859  HL2714      Tablet  ...      Curitiba           1600        NaN
715        9876  HL4379   Televisão  ...      Curitiba           2500        NaN
716        9918  HL1148      Câmera  ...      Curitiba           2100        NaN
717        9944  HL1918      iPhone  ...      Curitiba           5300        NaN
718        9998  HL7348  SmartWatch  ...      Curitiba           1400        NaN

[9999 rows x 10 columns]

In [51]:
tabela_total = pd.DataFrame()
for arquivo in lista_arquivo:
    if "Vendas" in arquivo:
        tabela = pd.read_csv(f"/content/drive/MyDrive/Vendas/{arquivo}") 
        tabela_total = tabela_total.append(tabela)
        print(arquivo)
display(tabela_total)



Vendas - Porto Alegre.csv
Vendas - Fortaleza.csv
Vendas - Salvador.csv
Vendas - Rio de Janeiro.csv
Vendas - São Paulo.csv
Vendas - Belo Horizonte.csv
Vendas - Recife.csv
Vendas - Goiás.csv
Vendas - Curitiba.csv


Unnamed: 0     SKU     Produto  ...          Loja Preco Unitario Unnamed: 8
0            14  HL4379   Televisão  ...  Porto Alegre           2500        NaN
1            34  HL1148      Câmera  ...  Porto Alegre           2100        NaN
2            43  HL1148      Câmera  ...  Porto Alegre           2100        NaN
3            49  HL8851    Notebook  ...  Porto Alegre           3500        NaN
4            52  HL4379   Televisão  ...  Porto Alegre           2500        NaN
..          ...     ...         ...  ...           ...            ...        ...
714        9859  HL2714      Tablet  ...      Curitiba           1600        NaN
715        9876  HL4379   Televisão  ...      Curitiba           2500        NaN
716        9918  HL1148      Câmera  ...      Curitiba           2100        NaN
717        9944  HL1918      iPhone  ...      Curitiba           5300        NaN
718        9998  HL7348  SmartWatch  ...      Curitiba           1400        NaN

[9999 rows x 10 columns]

In [52]:
#Passo 3 - Tratar/Compilar as bases de dados

display(tabela_total)


Unnamed: 0     SKU     Produto  ...          Loja Preco Unitario Unnamed: 8
0            14  HL4379   Televisão  ...  Porto Alegre           2500        NaN
1            34  HL1148      Câmera  ...  Porto Alegre           2100        NaN
2            43  HL1148      Câmera  ...  Porto Alegre           2100        NaN
3            49  HL8851    Notebook  ...  Porto Alegre           3500        NaN
4            52  HL4379   Televisão  ...  Porto Alegre           2500        NaN
..          ...     ...         ...  ...           ...            ...        ...
714        9859  HL2714      Tablet  ...      Curitiba           1600        NaN
715        9876  HL4379   Televisão  ...      Curitiba           2500        NaN
716        9918  HL1148      Câmera  ...      Curitiba           2100        NaN
717        9944  HL1918      iPhone  ...      Curitiba           5300        NaN
718        9998  HL7348  SmartWatch  ...      Curitiba           1400        NaN

[9999 rows x 10 columns]

In [67]:
# Calcular o produto mais vendido (em quantidade)
tabela_produtos = tabela_total.groupby("Produto").sum()
tabela_produtos = tabela_produtos[["Quantidade Vendida"]].sort_values(by="Quantidade Vendida", ascending= False)
display(tabela_produtos)

Quantidade Vendida
Produto                       
iPhone                    8974
Televisão                 5931
Notebook                  3249
Android                   3183
SmartWatch                2980
Tablet                    2921
Câmera                    2805

In [56]:
# Passo 5-Calcular o produto que mais faturou (em faturamento)
tabela_total["Faturamento"]=tabela_total["Quantidade Vendida"]* tabela_total["Preco Unitario"]

tabela_faturamento = tabela_total.groupby("Produto").sum()
#tabela_faturamento = tabela_faturamento[["Faturamento"]].sort_values(by="Faturamento", ascending=False)
display(tabela_faturamento)


Unnamed: 0  Quantidade Vendida  ...  Unnamed: 8  Faturamento
Produto                                     ...                         
Android        5164013                3183  ...         0.0     10822200
Câmera         4848296                2805  ...         0.0      5890500
Notebook       5393100                3249  ...         0.0     11371500
SmartWatch     4968367                2980  ...         0.0      4172000
Tablet         4920118                2921  ...         0.0      4673600
Televisão      9675708                5931  ...         0.0     14827500
iPhone        15015399                8974  ...         0.0     47562200

[7 rows x 5 columns]

In [57]:
#Produto que mais faturou em ordem crescente
tabela_faturamento = tabela_faturamento[["Faturamento"]].sort_values(by="Faturamento", ascending=False)
display(tabela_faturamento)

Faturamento
Produto                
iPhone         47562200
Televisão      14827500
Notebook       11371500
Android        10822200
Câmera          5890500
Tablet          4673600
SmartWatch      4172000

In [26]:
#Passo 6-Calcular a loja/cidade que mais vendeu (em faturamento) - criar um gráfico/dashboard

tabela_lojas = tabela_total.groupby("Loja").sum()
tabela_lojas = tabela_lojas[["Faturamento"]]
display(tabela_lojas)

Faturamento
Loja                       
Belo Horizonte      6480300
Curitiba            7060500
Fortaleza          14087900
Goiás               7441800
Porto Alegre        6868600
Recife              7303000
Rio de Janeiro     14867800
Salvador           13111300
São Paulo          22098300

In [71]:
import plotly.express as px
grafico = px.bar(tabela_lojas, x= tabela_lojas.index, y="Faturamento")
grafico.show()

In [68]:
import plotly.express as px
grafico = px.bar(tabela_produtos, x= tabela_produtos.index, y="Quantidade Vendida")
grafico.show()

In [69]:
import plotly.express as px
grafico = px.bar(tabela_faturamento, x= tabela_faturamento.index, y="Faturamento")
grafico.show()

# Conclusão

INSIGHTS:

ENTENDO OS ARQUIVOS DA BASE DE DADOS, PERCORREMOS TODOS OS ARQUIVOS E SEPARMOS OS ARQUIVOS "VENDAS" PARA CONSEGUIR IDENTIFICAR A SITUAÇÃO ATUAL DO DEPTO DE VENDAS E SEUS PRODUTOS EM LOJAS/CIDADES BEM COMO O NUMERO DE FATURAMENTO DE CADA PRODUTO EM COMPARAÇÃO A QUANTIDADE DE PRODUTOS VENDIDOS.


*   1º PRODUTO MAIS VENDIDO: IPHONE COM 8974 PRODUTOS VENDIDOS;
*   1º PRODUTO QUE MAIS GEROU FATURAMENTO: IPHONE COM $47.5622M DE FATURAMENTO;
*   1º CIDADE/LOJA QUE MAIS GEROU FATURAMENTO: SÃO PAULO COM 22.0983M
*   2º E 3º LUGAR DE PRODUTOS MAIS VENDIDOS E QUE GERAM MAIS FATURAMENTO TEMOS A TELEVISÃO E O NOTEBOOK.

COM ISSO, PODEMOS ENTENDER QUAIS PRODUTOS MENOS TROUXERAM RETORNO EM COMPARAÇÃO AOS DEMAIS, E QUAIS SÃO AS CIDADES QUE MAIS GERAM FATURAMENTO, DIRECIONANDO CAMPANHAS ESPECIFICAS E QUANTIDADE DE ESTOQUE PARA OS DETERMINADOS PRODUTOS E CIDADES.
